### PR TITLE
[Vue] migrate angularjs controllers + twig code to vue

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR depends on the core PR https://github.com/matomo-org/matomo/pull/19012.**

Changes:
* Migrate the ManagecustomalertsController to two new Vue components, ListAlerts and EditAlert

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by AltamashShaikh — CHANGES_REQUESTED** [review]
@diosmosis 

On Edit if additional emails are not set it generates an error and unable to edit any report.
Screenshot from 2022-04-12 05-37-31

additionalEmails is no more passed in the XHR request

**Comment by diosmosis** [comment]
@AltamashShaikh should be fixed

**Review by AltamashShaikh — APPROVED** [review]
@diosmosis 

On Edit if additional emails are not set it generates an error and unable to edit any report.
Screenshot from 2022-04-12 05-37-31

additionalEmails is no more passed in the XHR request

**Comment by diosmosis** [comment]
@AltamashShaikh the comment posted is the same as before, just to be clear, it's fixed which is why you approved it right?

File: vue/src/ListAlerts/ListAlerts.vue
Comment (sgiehl) [inline vue/src/ListAlerts/ListAlerts.vue]: The sitename is being displayed encoded

File: vue/src/EditAlert/EditAlert.vue
Comment (sgiehl) [inline vue/src/EditAlert/EditAlert.vue:23]: The preselected value is also displayed encoded here

**Comment by diosmosis** [comment]
@sgiehl :+1: should be fixed

File: vue/src/ListAlerts/ListAlerts.vue
Comment (sgiehl) [inline vue/src/ListAlerts/ListAlerts.vue]: Here we could use translated `Intl_PeriodDay`, `Intl_PeriodMonth`, ... instead, so it's displayed translated

**Comment by sgiehl** [comment]
Besides the comment above that one looks good to me.

**Comment by sgiehl** [comment]
@diosmosis we also should already raise the matomo requirement in plugin.json, so we can't forget it when releasing the plugin

**Comment by diosmosis** [comment]
@sgiehl done

**Review by sgiehl — APPROVED** [review]
Looks good to me now. UI tests might need an update though.

@AltamashShaikh feel free to merge that one when it fits into the release process.

**Comment by diosmosis** [comment]
@sgiehl does that mean I can't merge it now?

**Comment by sgiehl** [comment]
I don't know if any of the other PRs should be merged and released before.

**Comment by diosmosis** [comment]
@sgiehl this is due to potentially merging changes before the version bump? Would this apply to LogViewer as well or just CustomAlerts since there may be another scheduled change?

**Comment by sgiehl** [comment]
I didn't have a look at the open prs here... If there aren't any for LogViewer I guess it should be fine to merge there already.

**Comment by diosmosis** [comment]
@tsteur would it be possible to get some guidance on how to merge these plugin Vue PRs, given merging is required for milestone completion?

**Comment by AltamashShaikh** [comment]
@diosmosis Just an update on release process, we do a release on 2nd and 4th Monday. Generally the plugins are good to merge once we get a approval on the PR. Since we have updated the min requirement for it should be merged only after we release the min core version.
Also for your reference, even though we can merge this PR now but cloud customers can get this update only after release of 4.10.0 on cloud.

@tsteur should we wait for 4.10.0 to be released on cloud in such cases ?

**Comment by diosmosis** [comment]
@AltamashShaikh :+1: thanks for the clarification

File: plugin.json
Comment (tsteur) [inline plugin.json]: this should be 4.1.0?

**Comment by tsteur** [comment]
@AltamashShaikh we can release this already now. Because Matomo 4.10 is required, the Cloud team will automatically only be able to upgrade to this version once 4.10 has been released. 

@AltamashShaikh @diosmosis there is currently version number 5.0 configured, I believe this is an error as we always maintain the same major version number for plugins as core. It should be 4.1.0

File: plugin.json
Comment (diosmosis) [reply plugin.json]: Seemed like a major change, I'll do minor bumps instead.

Comment (tsteur) [reply plugin.json]: 👍  We always keep major version the same as with core. I assume no functionality changes anyway and there are no breaking changes etc so a minor version increase is best anyway.

**Pull request merged**
